src/pykeen/datasets/ea/combination.py

```diff
@@ -44,69 +226,172 @@ def __call__(
             a single triples factory comprising the joint graph, as well as a tensor of pairs of matching IDs.
             The tensor of matching pairs has shape `(2, num_alignments)`, where `num_alignments` can also be 0.
         """
+        # concatenate triples
+        mapped_triples, offsets = cat_shift_triples(left, right)
+        # filter alignment and translate to IDs
+        alignment = filter_map_alignment(alignment=alignment, left=left, right=right, entity_offsets=offsets[:, 0])
+        # process
+        mapped_triples, alignment, translation_kwargs = self._process(mapped_triples, alignment, offsets)
+        if isinstance(left, TriplesFactory) and isinstance(right, TriplesFactory):
+            # merge mappings
+            entity_to_id, relation_to_id = merge_label_to_id_mappings(
+                # TODO: offsets can also be dense mapping
+                # translation: Union[torch.LongTensor, Tuple[torch.LongTensor, torch.LongTensor]]
+                left=left,
+                right=right,
+                **translation_kwargs,
+            )
+            triples_factory = TriplesFactory(
+                mapped_triples=mapped_triples,
+                entity_to_id=entity_to_id,
+                relation_to_id=relation_to_id,
+                **kwargs,
+            )
+        else:
+            triples_factory = CoreTriplesFactory(mapped_triples=mapped_triples, **kwargs)
+
+        return triples_factory, alignment
+
+    @abstractmethod
+    def _process(
```

Comment:
this function needs a full docstring, probably should be public since it's the one that needs to get overridden